src/http/mod.rs

```diff
@@ -157,6 +158,93 @@ impl ObjectStore for HttpStore {
         .boxed()
     }
 
+    fn list_opts(
```

Comment:
This implementation seems over complicated given that webdav doesn't support pagination.

Could we perhaps reuse more of the existing logic?

Reply:
yeah, i considered reuse the logic of `list` and `list_with_delimiter`, but since the response of `list_opts` is a BoxStream with static lifecycle, so cannot reuse  `list_with_delimiter` directly(please correct me if misunderstand something).